### PR TITLE
Flow bytes pkts syntax/v3

### PR DESCRIPTION
**Pull request by inashivb**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/5646

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2079

1. Syntax comparisons:

| master    | This PR |
| -------- | ------- |
| flow.pkts_toclient  | flow.pkts:toclient    |
| flow.pkts_toserver | flow.pkts:toserver     |
| flow.bytes_toserver   | flow.bytes:toserver    |
| flow.bytes_toclient | flow.bytes:toclient |
| - | flow.pkts:either |
| - | flow.bytes:either |

2. Code advantages shared in the commit message of ac9c5d2bf212d7cd8e36a01b4ff0bb1a0257e6c2

Previous PR: https://github.com/OISF/suricata/pull/11653
Changes since v1:
- rebased and made PR description clearer so difference is easy to see as asked

Known issues:
1. Doc update hasn't been done yet.
2. Prefilter fn has not been updated yet so it may not work as intended.
4. Tests for the parsing of arguments need to be done.

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23002

**Comment by victorjulien** [comment]
```
detect-flow-pkts.c:44:17: warning: The left operand of '==' is a garbage value [core.UndefinedBinaryOperatorResult]
   44 |     if (df->dir == DETECT_FLOW_TO_SERVER) {
      |         ~~~~~~~ ^
1 warning generated.
```

**Comment by victorjulien** [comment]
The existing keywords map 1:1 to the flow.bytes_toclient and friends fields from the eve record. Wonder if we should keep them as is.

File: src/detect-flow-pkts.c [hunk 1]
```diff
@@ -23,114 +23,103 @@
...
-    sigmatch_table[DETECT_FLOW_PKTS_TO_CLIENT].SetupPrefilter = PrefilterSetupFlowPktsToClient;
-}
-
-static int DetectFlowPktsToServerMatch(
-        DetectEngineThreadCtx *det_ctx, Packet *p, const Signature *s, const SigMatchCtx *ctx)
+static int DetectFlowPktsSetup(DetectEngineCtx *de_ctx, Signature *s, const char *rawstr)
 {
-    if (p->flow == NULL) {
-        return 0;
+    DetectFlow *df = SCCalloc(1, sizeof(DetectFlow));
```
Comment (victorjulien) [inline src/detect-flow-pkts.c:70]: no check after alloc

**Comment by inashivb** [comment]
> The existing keywords map 1:1 to the flow.bytes_toclient and friends fields from the eve record. Wonder if we should keep them as is.

Do you mean keep the rule language as-is? Sure! I just need a verdict on this so I can implement the `either` option as is better suited. :)

**Pull request closed without merging**